src/pykoala/ancillary.py

```diff
@@ -14,21 +14,54 @@
 # =============================================================================
 # Astropy and associated packages
 # =============================================================================
-from astropy.io import fits
+#from astropy.io import fits
 # =============================================================================
-
 # =============================================================================
 # PyKOALA modules
 # =============================================================================
 from pykoala.cubing import Cube, build_wcs, build_cube
-
-
+# =============================================================================
+# =============================================================================
+# =============================================================================
+# ANGEL modules                                                            #!!!
+# =============================================================================
+# =============================================================================
+# =============================================================================
 def vprint(*arg, **kwargs):
     """
     Prints the arguments only if verbose=True.
     """
-    if kwargs['verbose']:
-        print(*arg)
+    try:
+        if kwargs['verbose']:
+            print(*arg)
+    except Exception:
+        do_nothing = 0                    #!!! Ángel: Una ayudita aquí... si no quiero que haga nada si falla, ¿qué pongo?
```

Comment:
Try this
`
kwargs.get('verbose', False)
if verbose:
    print(*args)
`

Reply:
A similar function is implemented on the Correction base class, so it might be a good idea to refactor the pipeline and always use the same function.

Reply:
As I said below, we should probably use `log` everywhere (or never)

Reply:
I don't know (yet) how to use `log`.  I can't learn everything at your level in 2 days!

The similar function in Correction base class is a part of the class (self):

```
def corr_print(self, msg, *args):
        """Print a message."""
        if self.verbose:
            print("[Correction: {}] {}".format(self.name, msg), *args)
```

After several testings, replaced for:

```
def vprint(*args, verbose=False, **kwargs):
    if verbose: print(*args)
```

By default verbose=False (unless specifically asked, do not print anything). It needs the **kwargs as verbose many times there, plus accepting other kwargs (i.e., for avoiding  `TypeError: vprint() got an unexpected keyword argument 'plot'` ).

I could remove  **kwargs if using: `vprint(' ', verbose=kwargs.get("verbose"))`, but this fails if verbose not given in kwargs...

Reply:
Your suggestion `def vprint(*args, verbose=False, **kwargs):` is totally fine. A possible alternative using `get` would be
```
def vprint(*args, **kwargs):
    if kwargs.get("verbose", False):
        print(*args)
```
(the second argument to `get` is the default value, if the requested key is not present)